internal/charmstore/stats.go

```diff
@@ -161,10 +171,21 @@ func timeToStamp(t time.Time) int32 {
 // IncCounterAsync increases by one the counter associated with the composed
 // key. The action is done in the background using a separate goroutine.
 func (s *Store) IncCounterAsync(key []string) {
-	go func() {
+	s.Go(func(s *Store) {
 		if err := s.IncCounter(key); err != nil {
 			logger.Errorf("cannot increase stats counter for key %v: %v", key, err)
 		}
+	})
+}
+
+// Go runs the given function in a new goroutine,
+// passing it a copy of s, which will be closed
+// after the function returns.
```

Comment:
I was going to ask "why the s.Go(...) calls?"  Now I see the benefit. Really nice.